adapters/dmx/dmx.go

```diff
@@ -327,3 +339,17 @@ func isDmxParams(t interface{}) bool {
 		return false
 	}
 }
+
+func getIdfa(request *openrtb.BidRequest) (string, bool) {
+	device := request.Device
+	if device.IFA != "" {
+		return device.IFA, true
+	}
+	return "empty", false
+}
+func checkProtocols(imp *openrtb.Video) []openrtb.Protocol {
+	if len(imp.Protocols) > 0 {
+		return imp.Protocols
+	}
+	return []openrtb.Protocol{2, 3, 5, 6, 7, 8}
```

Comment:
Small Optimization: Please consider defining the protocol list at the package level. It's safe because it's read only within the adapter. This will slightly reduce memory overheard and slightly improve performance.